internal/querycoordv2/balance/rowcount_based_balancer.go

```diff
@@ -67,7 +67,8 @@ func (b *RowCountBasedBalancer) AssignSegment(segments []*meta.Segment, nodes []
 
 func (b *RowCountBasedBalancer) convertToNodeItems(nodeIDs []int64) []*nodeItem {
 	ret := make([]*nodeItem, 0, len(nodeIDs))
-	for _, node := range nodeIDs {
+	for _, nodeInfo := range b.getNodes(nodeIDs) {
```

Comment:
Why not use nodeIDs directly?

Reply:
It will filter the dropping node in the `getNodes ` method.